src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -724,8 +710,14 @@ private async Task<IVsWindowFrame> CreateDocWindowForSolutionAsync()
                 return null;
             }
 
-            // pass empty array of NuGetProject
-            var uiController = UIFactory.Value.Create(projects);
+            var projectContexts = new List<ProjectContextInfo>(projects.Count());
+            foreach (var project in projects) // TODO: Optimize # of calls to server
```

Comment:
Above, why not use `INuGetProjectManagerService.GetProjectsAsync(...)` instead of `ISolutionManager.GetNuGetProjectsAsync()`?